test/extensions/network/dns_resolver/cares/dns_impl_test.cc

```diff
@@ -2089,5 +2089,35 @@ TEST_P(DnsImplCustomResolverTest, CustomResolverValidAfterChannelDestruction) {
   EXPECT_FALSE(peer_->isChannelDirty());
 }
 
+class DnsImplAresFlagsForMaxUdpQueriesinTest : public DnsImplTest {
+protected:
+  bool tcpOnly() const override { return false; }
+  void updateDnsResolverOptions() override {
+    auto udp_max_queries = std::make_unique<ProtobufWkt::UInt32Value>();
+    udp_max_queries->set_value(100);
+    dns_resolver_options_.set_allocated_udp_max_queries(
+        dynamic_cast<ProtobufWkt::UInt32Value*>(udp_max_queries.get()));
+  }
+};
+
+// Parameterize the DNS test server socket address.
+INSTANTIATE_TEST_SUITE_P(IpVersions, DnsImplAresFlagsForMaxUdpQueriesinTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// Validate that c_ares flag `ARES_OPT_UDP_MAX_QUERIES` is set when UInt32 property
+// `udp_max_queries` is set.
+TEST_P(DnsImplAresFlagsForMaxUdpQueriesinTest, UdpMaxQueriesIsSet) {
```

Comment:
nit:
Is there a way to verify the UdpMaxQueries number actually is enforced by the c-ares library?

Reply:
@yanjunxiang-google - do you mean by checking opts.udp_max_queries or something more involved ? I tried poking in that area but couldn't find anything obvious - if you have suggestions - happy to look into it!

Reply:
I was pushing c'ares for this change here: https://github.com/c-ares/c-ares/issues/444

I believe you can test like this:
- set this value as n 
- keep pumping traffic. 
- Dont turn on dns over tcp
- Sniffing ('ngrep -W single -l -q -d any -i "" udp port 53') the DNS packets
- The src port should keep changing after every n requests.

Reply:
This change should be ok as is, since we rely on c-ares to test the behavior of the ARES_OPT_UDP_MAX_QUERIES flag. I do not think it is important to validate that this flag is working in c-ares.